cmd/kube-apiserver/app/server.go

```diff
@@ -523,6 +525,10 @@ func buildGenericConfig(
 		lastErr = fmt.Errorf("failed to initialize admission: %v", err)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIPriorityAndFairness) {
```

Comment:
```suggestion
	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIPriorityAndFairness) {
		_, flowcontrolAPIEnabled := s.APIEnablement.RuntimeConfig["flowcontrol.apiserver.k8s.io"]
		_, allAPIEnabled := s.APIEnablement.RuntimeConfig[resourceconfig.APIAll]
		_, alphaAPIEnabled := s.APIEnablement.RuntimeConfig[resourceconfig.APIAlpha]
		switch {
		case !s.GenericServerRunOptions.EnableInflightQuotaHandler:
		case !flowcontrolAPIEnabled && !alphaAPIEnabled && !allAPIEnabled:
		default:
			genericConfig.FlowControl = BuildPriorityAndFairness(s, clientgoExternalClient, versionedInformers)
		}
	}
```
there's currently 3 conditions for enabling the feature: (a) the `--enable-inflight-quota-handler` flag (b) the feature-gate (c) the API enablement. we can simply check all these conditions at this point when constructing the flowcontrol system so that we only check if the system instance is nil on deciding whether to replace the legacy handler w/ the new one.

Reply:
I do not clearly remember the reason that the `--enable-inflight-quota-handler` flag was introduced.  I have a vague memory that it was about switching the config source from max-mutating-inflight + max-readonly-inflight to a single number.

It seems silly to add a flag whose purpose is to enable the API Priority and Fairness feature.  That's what feature gates are already there for.

Reply:
> I do not clearly remember the reason that the --enable-inflight-quota-handler flag 

Possibly at my request, since the feature flag is temporary and the need to optionally not run this is permanent.

Reply:
At any rate the suggested change here should go in a different PR I think where we can debate it separately.